### DISCUSSION TITLE
Failed to collect page data for /api/webhook/clerk

### DISCUSSION THREAD

**Discussion by mishra-gaurav-04** [vercel/next.js#59743] [Help]
### Summary

I have a next js application it's api routes are working in dev mode but on deploying it on vercel it is giving me an error of Failed to collect page data for /api/webhook/clerk ,I am using prisma ORM for handing the crud operation and mongo db is being used as database. Any suggestion regarding this

![Screenshot from 2023-12-19 00-33-36](https://github.com/vercel/next.js/assets/108576654/88c8aa6a-4568-463f-ba1c-10ed628eec16)

Any suggestion regarding this will be very helpful
Thanks in advance

**Comment by cintra1** [comment]
It's very simple, add 'use client' to your route.ts or react-redux.ts file.

**Reply by gaurangrshah** [reply]
You can add the "use client" directive to api routes? What does that do?

**Reply by cintra1** [reply]
In fact, 'use client' marks the module and its transitive dependencies as client code, so it doesn't make sense to add the code to a root file. However, in my case, it corrected my deployment error in Vercel and then I removed the 'use client' and it continued to work, it was more like a workaround.

**Reply by gaurangrshah** [reply]
Interesting, will give it  a try - thanks!
